tools/i18n-plugin/src/main/java/org/openhab/core/tools/i18n/plugin/BundleInfoReader.java

```diff
@@ -102,11 +103,20 @@ private void readThingInfo(Path ohinfPath, BundleInfo bundleInfo) throws IOExcep
                 }
                 for (Object type : types) {
                     if (type instanceof ThingTypeXmlResult) {
-                        bundleInfo.getThingTypesXml().add((ThingTypeXmlResult) type);
+                        ThingTypeXmlResult result = (ThingTypeXmlResult) type;
+                        bundleInfo.getThingTypesXml().add(result);
+                        if (bundleInfo.getBindingId().isBlank()) {
+                            bundleInfo.setBindingId(result.getUID().getBindingId());
+                        }
                     } else if (type instanceof ChannelGroupTypeXmlResult) {
-                        bundleInfo.getChannelGroupTypesXml().add((ChannelGroupTypeXmlResult) type);
+                        ChannelGroupTypeXmlResult result = (ChannelGroupTypeXmlResult) type;
+                        bundleInfo.getChannelGroupTypesXml().add(result);
+                        if (bundleInfo.getBindingId().isBlank()) {
+                            bundleInfo.setBindingId(result.getUID().getBindingId());
+                        }
                     } else if (type instanceof ChannelTypeXmlResult) {
-                        bundleInfo.getChannelTypesXml().add((ChannelTypeXmlResult) type);
+                        ChannelTypeXmlResult result = (ChannelTypeXmlResult) type;
+                        bundleInfo.getChannelTypesXml().add(result);
```

Comment:
What about setting the `bindingId` here in the `bundleInfo` too?

Reply:
These kind of XML files don't have a `bindingId` attribute which is used in a `UID`. The parsed objects only have `URI`s which can be based on anything.

Reply:
I see the `ChannelType` does extend from `AbstractDescriptionType` that has a `UID`, so I'll check if I can add a bindingId based on that. :slightly_smiling_face: